shelf/endpoint_decorators.py

```diff
@@ -214,18 +218,20 @@ def decode_request(self, container):
             Returns:
                 object | None: decoded JSON from request. None if invalid.
         """
-        data = container.request.get_json(silent=True, force=True)
+        if container.request.data:
+            data = container.request.get_json(silent=True, force=True)
 
-        if not isinstance(data, (list, dict)):
-            container.context.add_error(ErrorCode.INVALID_REQUEST_DATA_FORMAT)
-            data = None
+            if not isinstance(data, (list, dict)):
+                container.context.add_error(ErrorCode.INVALID_REQUEST_DATA_FORMAT)
+                data = None
+        else:
+            data = deepcopy(default_request_body)
```

Comment:
Can you comment on why we are deepcopying this? I can see it getting removed. I had no clue what it was for immediately.